pkg/ccl/workloadccl/fixture.go

```diff
@@ -647,40 +646,44 @@ func RestoreFixture(
 			}
 			if len(resCols) == 7 {
 				if err := res.Scan(
-					&discard, &discard, &discard, &rows, &index, &discard, &tableBytes,
+					&discard, &discard, &discard, &rows, &index, &discard, &discard,
+				); err != nil {
+					return err
+				}
+			} else if len(resCols) == 6 {
```

Comment:
This is the expected result size going forward, right? Then, it seems the above case would benefit from a version gate, since we presumably don't want to keep backward compatibility indefinitely? 

CC @cockroachdb/disaster-recovery

Reply:
It's to preserve backwards compatibility so you can use `cockroach workload` built on master to load data into an older version. This is useful for e.g. mixed version tests since figuring out the matching binary needed is cumbersome and instead you can just use master.

That being said, only two tests use `fixtures load` and neither are mixed version tests. If we want to remove it we won't break any tests and backwards compatibility for workload doesn't appear to be a hard rule.

Reply:
> It's to preserve backwards compatibility so you can use cockroach workload built on master to load data into an older version. 

Interesting, I didn't realize we were relying on that. That's inherently error-prone. I get the convenience aspect, but sooner or later we'll run into other compatibility issues. I think we should switch to using the same versioned binary (for workloads) in mixed version tests.

> If we want to remove it we won't break any tests and backwards compatibility for workload doesn't appear to be a hard rule.

The side-effect of adding special cases (without version gates) for backward compatibility is they tend to pollute the rest of the code; i.e., maintenance over time gets trickier since 6 months from now it won't be as clear as to why there is a special case there.

Reply:
> That's inherently error-prone. I get the convenience aspect, but sooner or later we'll run into other compatibility issues

Agreed, [we've already run into compatibility issues due to newly added flags](https://github.com/cockroachdb/cockroach/blame/87b463a09f714af53270f4307d38a01fc76cebf6/pkg/cmd/roachtest/tests/admission_control_elastic_mixed_version.go#L78-L82). We should have the helper expose the binary path of the currently running version.

>I think we should switch to using the same versioned binary 

Need to think about this more, but some tests spin up a workload (e.g. tpcc) in the background, and have it run throughout the entire test uninterrupted.

> The side-effect of adding special cases (without version gates) for backward compatibility is they tend to pollute the rest of the code

Sounds like we're leaning towards removing the backwards compatibility checks then? If so I'm for it, makes it easier for me too :)

Reply:
> Need to think about this more, but some tests spin up a workload (e.g. tpcc) in the background, and have it run throughout the entire test uninterrupted.

Right. It's definitely a convenience feature to keep pumping some form of background traffic, continuously. Synchronizing workload binaries with cockroach binaries seems tricky; would require a non-trivial change to the mixedversion framework.